app/src/main/java/com/univoice/feature/signup/SchoolInputActivity.kt

```diff
@@ -146,6 +149,14 @@ class SchoolInputActivity :
         adapter.submitList(filteredList)
     }
 
+    private fun observeSchoolList() {
+        lifecycleScope.launch {
+            viewModel.schoolList.collect { schoolList ->
+                filterSchools(binding.etSchoolInputSearch.text.toString().trim())
+            }
+        }
+    }
+
```

Comment:
P2 : UiState를 통해 이렇게 서버 통신을 하면서 발생하는 케이스별을 관리할 수 있어요!
```suggestion
    private fun setupUniversityObserve() {
        viewModel.postUniversityState.flowWithLifecycle(lifecycle).onEach {
            when (it) {
                is UiState.Loading -> Unit
                is UiState.Success -> {
                    // 여기에서 받아온 리스트 데이터 연결해주기! it.data에 담겨있어용
                }

                is UiState.Empty -> Unit
                is UiState.Failure -> Timber.tag("SchoolInputActivity").d(it.msg)
            }
        }.launchIn(lifecycleScope)
    }

```